### ISSUE TITLE
Crash

### ISSUE THREAD

**Issue by asadsam** [aleemrazzaq/iOSCSR#1]
@aleemrazzaq Hi there, while trying to run the app after integrating this framework, the app builds successfully but crashes just after running, giving the below log in console. Could you please take a look into it and help resolve.
Much appreciated, thanks.

dyld: Library not loaded: @rpath/libswiftCore.dylib
Referenced from: /private/var/containers/Bundle/Application/6043265D-17E1-4C32-89E2-B523786C0267/CreateCSR_RazzECC.app/Frameworks/CryptoAPI.framework/CryptoAPI
 Reason: image not found

**Comment by aleemrazzaq** [comment]
Go to
Target Settings of your App then Build Setting then set **Always Embed Swift Standard Libraries** to **NO**
<img width="604" alt="screen shot 2017-10-12 at 6 10 35 pm" src="https://user-images.githubusercontent.com/5242797/31497601-03ee47c8-af79-11e7-9d2f-fab840c50e8d.png">
Try and let me know

**Comment by asadsam** [comment]
@aleemrazzaq it is already set to No, brother. Cross checked again and run the app, the same error.
Could you please help me with a working example too, thanks.

**Comment by aleemrazzaq** [comment]
@asadsam my mistake, it should be YES. You may get help from this thread https://stackoverflow.com/questions/24333981/ios-app-with-framework-crashed-on-device-dyld-library-not-loaded-xcode-6-beta
try and let me know.

**Comment by asadsam** [comment]
@aleemrazzaq  Referred the above link, nothing worked but only changing the framework status from required to optional worked. The app isn't crashing now but all the values are nil for KeyPairAttributes and KeyPairManager.

**Comment by aleemrazzaq** [comment]
Its working 100% fine on my side, and I generated accurately CSR on my side
I already mentioned how to use. like following

NSString * currentlLoginEmailAddress = @"test102@test.com"; // Unique string use as alias for Keypairs
  NSString * bundleID = @"com.myCompany.myApp"; // Bundle ID of your App use as Application Tag
  NSString * userName = @"user name"; // User Name or name for organization which use in Certificate
  NSString * randomNumber = @"453963013"; // Any Serial number which use in Certificate.

  KeyPairAttributes * keyPairAttributes = [[KeyPairAttributes alloc] init];
  keyPairAttributes.keyLabel = currentlLoginEmailAddress;
  keyPairAttributes.keySize = KeyPairAttributes.keySize256;
  keyPairAttributes.keyStorageType = KeyPairAttributes.Hardware;
  keyPairAttributes.keyType = KeyPairAttributes.EC;
  keyPairAttributes.keyApplicationTag = bundleID;

  KeyPairManager * keypairManager = [[KeyPairManager alloc] init];
  //delete old keypair with same alias if exist.
  [keypairManager deleteKeyPairWithKeyTag:currentlLoginEmailAddress];
  //generate keypair from keypair attributes
  [keypairManager generateKeyPair:keyPairAttributes completionHandler:^(OSStatus status) {
  //get private key with alias and prompt message that will appear in Touch ID dialog.
      SecKeyRef privateKey = [keypairManager getPrivateKeyWithKeyTag:currentlLoginEmailAddress prompotMessage:@""];
      if (privateKey != nil) {
    // get public in data format 
        NSData * publicKeyData = [keypairManager getPublicKeyWithKeyTag:currentlLoginEmailAddress];
        
        SubjectDN * subjectDN = [[SubjectDN alloc] init];
        subjectDN.commonName = userName;
        subjectDN.serialNumber = randomNumber;
        subjectDN.emaiAddress = currentlLoginEmailAddress;
        // generate certificate with all subject with Public key in data and private key in seckeyreference format
        NSString * csr = [[CertificateManager getInstance] generateCertificateSigningRequestWithSubjectDN:subjectDN publicKeyData:publicKeyData privateKey:privateKey];
        if (csr.length > 0) {
        NSLog(@"%@",csr);
        }
    }
}];

**Comment by asadsam** [comment]
@aleemrazzaq.. hey, i think it was a problem with xcode 8.3.1. I have installed xcode 9 and the crash has disappeared fortunately.
But i am getting this following error while generating key pairs, i am using the same code which you have mentioned. Could you please help me find the cause of this error. Thanks.

The error is:
Failed to generate key pairs
Error getting private key data: -25300

**Comment by aleemrazzaq** [comment]
@asadsam on which device you are testing ?
check this status on osstatus.com 
https://www.osstatus.com/search/results?platform=all&framework=all&search=-25300

**Comment by asadsam** [comment]
@aleemrazzaq Hi bro, thanks for all the help. Its working perfectly fine with xcode 9(latest) and iphone 7 plus with ios 11(latest).

**Comment by aleemrazzaq** [comment]
Yes it should because I tested this before uploading, and will not work on simulator and those devices which hasn't touch id

**Comment by asadsam** [comment]
Great!! Are you planning make it open source? For others also to contribute. Thanks again.

**Comment by aleemrazzaq** [comment]
Yeah sure, soon I'll. I already spend too much time to solve this issue because Apple only support 256 with EC in secure enclave and also there wasn't too much help available online when I was doing R&D on this.

**Comment by asadsam** [comment]
@aleemrazzaq Hi, hope you are doing good. 
I had one more doubt bro, your kind help is needed. I am trying to use ECC key size 521 (kSecp521r1), how would i be able to achieve that in this framework? 
(I see only 2 options to set the keysize that are 256 and 512)

**Comment by zealit** [comment]
can you also make it working on OS X?
